src/app/app.component.ts

```diff
@@ -13,7 +13,13 @@ export class AppComponent {
   constructor(authService: AuthService, cmsService: CmsService) {
     authService.token.subscribe((token) => {
       if (token) {
-        cmsService.setToken(token);
+        let refresher: any;
+        cmsService.setToken(token, cb => {
```

Comment:
The CMS and Auth services were purposefully decoupled.  But I needed some way for CMS to request that Auth "go get a new token".

This seemed like the best place to inject that callback function into CMS ... since this was already telling CMS about any new auth tokens.

I wish this was cleaner, but not sure I have a better idea for this bi-directional communication.